### PR TITLE
DOC: Polynomial documentation clarification for domain

### PR DESCRIPTION
**Pull request by StanczakDominik**
This is a short follow-up to my not-as-short rant on https://github.com/numpy/numpy/issues/9533. It clarifies a point for the Polynomial documentation.

### CODE REVIEW

File: numpy/polynomial/polynomial.py [hunk 1]
```diff
@@ -1443,6 +1443,10 @@ class Polynomial(ABCPolyBase):
     '+', '-', '*', '//', '%', 'divmod', '**', and '()' as well as the
     attributes and methods listed in the `ABCPolyBase` documentation.
 
+    Please note that ``Polynomial.coef`` are coefficients in the scaled
```
Comment (eric-wieser) [inline numpy/polynomial/polynomial.py]: Can we put this in the docs for `coef` instead?

Comment (StanczakDominik) [reply numpy/polynomial/polynomial.py]: I don't think that's going to work. In IPython:

```
[ins] In [18]: p = np.polynomial.Polynomial.fit(a, b, 1)                                                                                                                                                              
Out[18]: Polynomial([3.5, 0.5], domain=[0., 1.], window=[-1.,  1.])

[ins] In [20]: p.coef?                                                                                                                                                                                            
# docs for ndarray
[ins] In [21]: help(p.coef)
# likewise                            
```

Comment (eric-wieser) [reply numpy/polynomial/polynomial.py]: If nothing else, this docstring belongs in `ABCPolyBase` not here - it applies to all subclasses.

Comment (StanczakDominik) [reply numpy/polynomial/polynomial.py]: Sure, but when I, the Well Intentioned User intending to Write Code According To New Standards, look for documentation on this, I'm going to find [docs for Polynomial](https://docs.scipy.org/doc/numpy/reference/generated/numpy.polynomial.polynomial.Polynomial.html#numpy.polynomial.polynomial.Polynomial), not for `ABCPolyBase`; and given that `ABCPolyBase` is not actually linked from `Polynomial`, the odds of me going looking for them fall off pretty sharply.

I'd keep it here and add it to ABCPolyBase, if you don't mind.

A separate issue is that I only found that page after correcting for the fact that google prioritizes old numpy doc versions for some reason.

Comment (eric-wieser) [reply numpy/polynomial/polynomial.py]: Does the link on the line immediately above not work?

Comment (StanczakDominik) [reply numpy/polynomial/polynomial.py]: No, actually, if you take a look at the rendered docs I linked, it's not a hyperlink - it's formatted in bold, though.

Comment (StanczakDominik) [reply numpy/polynomial/polynomial.py]: No, actually, if you take a look at the rendered docs I linked, it's not a hyperlink - it's formatted in bold, though.

File: numpy/polynomial/polynomial.py [hunk 2]
```diff
@@ -1441,7 +1441,7 @@ class Polynomial(ABCPolyBase):
 
     The Polynomial class provides the standard Python numerical methods
     '+', '-', '*', '//', '%', 'divmod', '**', and '()' as well as the
-    attributes and methods listed in the `ABCPolyBase` documentation.
+    attributes and methods listed in the ``ABCPolyBase`` documentation.
```
Comment (eric-wieser) [inline numpy/polynomial/polynomial.py]: The link was correct as is - the problem is the thing its trying to link to doesn't exist!

File: numpy/polynomial/polynomial.py (hunk 1 above)
Comment (charris) [reply numpy/polynomial/polynomial.py]: The documentation for `p` is inherited from `ABCPolyBase`.

Comment (StanczakDominik) [reply numpy/polynomial/polynomial.py]: I tried it again:
```
>>>help(p)
Help on Polynomial in module numpy.polynomial.polynomial object:

class Polynomial(numpy.polynomial._polybase.ABCPolyBase)
 |  Polynomial(coef, domain=None, window=None)
 |  
 |  A power series class.
 |  
 |  The Polynomial class provides the standard Python numerical methods
 |  '+', '-', '*', '//', '%', 'divmod', '**', and '()' as well as the
 |  attributes and methods listed in the `ABCPolyBase` documentation.
 |  
 |  Parameters
 |  ----------
 |  coef : array_like
 |      Polynomial coefficients in order of increasing degree, i.e.,
 |      ``(1, 2, 3)`` give ``1 + 2*x + 3*x**2``.

```

I may be misunderstanding something, but if it were to be inheriting or copying ABCPolyBase's docstring, I'd expect to see the "attributes and methods listed in the `ABCPolyBase` documentation." line as seen in the change above.

**Comment by rossbar** [comment]
It's been a while since this was last discussed, but I think it would definitely be an improvement to have the note about the fact that the coefficients are in the scaled domain.

The main sticking point seems to be where/how to include the information. A couple thoughts on some things that could be done to resolve this:
 - The `coef` attribute could be turned into a property and thus given a docstring, so that when a user does `help(p.coef)`, they get the property docstring instead of the docstring for `ndarray`.
 - The note about coefficients could be added to the `Parameters` section for each class-level docstring (this seems to be in line with @StanczakDominik 's current proposal). The downside is that this is a lot of repetition and has the potential to fall out of sync, but it is in line with the way the subclass docstrings are currently done.
 - Alternatively, the docstrings for the subclasses could be made to re-use some or all of the docstring from `ABCPolyBase`. This would eliminate some repitition (the docstrings for each of the classes that derive from `ABCPolyBase` have a lot of the same information). One downside (at least if it is done via some form of manually setting `__doc__` in the subclasses) is that it might make the source code a little less readable, though the docstring when accessed via `help` or `?` etc. would be fine.

Feedback on these ideas would be most welcome. @StanczakDominik do you have the interest/bandwidth to keep pursuing this?

**Comment by rossbar** [comment]
Well I'm not sure there's a *right* solution, but picking this back up again definitely seems worthwhile to me. IMO the following is quite confusing and needs to be documented:

```python
>>> a = np.arange(10)
>>> b = a.copy()
>>> np.polynomial.Polynomial.fit(a, b, deg=1).coef
array([4.5, 4.5])
>>> np.polynomial.polynomial.polyfit(a, b, deg=1)
array([-9.67166378e-17,  1.00000000e+00])
```

Hopefully a workable solution can be settled on here - if a more substantial re-structuring of the docstrings ends up being desired, it can be tackled in subsequent PRs.

**Comment by StanczakDominik** [comment]
Hey, I completely forgot about this one. Let's see if this addition works...

**Comment by melissawm** [comment]
@rossbar can you take another look at this?

**Comment by melissawm** [comment]
Gentle ping @rossbar  :)

**Comment by vpmakinen** [comment]
I ended up here after running this line of code
```
fit = Polynomial.fit(x, y, deg=1)
print(fit)
>>> 8.503289002047184 - 0.8248715395742434 x**1
```
I definitely did not expect to see coefficients in any scaled domain and spent some time before realizing what is going on. If the scaling is not identity, maybe using some other letter than x or adding "(scaled)" somewhere would help users to notice this.

**Comment by melissawm** [comment]
Ping, @rossbar :)

**Pull request closed without merging**